Urgent fixes

kdaveed opened the pull request:
Auto label generation for images and bone segment selector acceleration for study design execution were implemented.

Commit:
Simplified navigator for faster selection.

Commit:
Now for each added Image a label with the value "ImageLabel"
is generated automatically.

zarquon42b commented:
How am I supposed to select a SI from a large list? There is no possibility to scroll!

![selection_137](https://cloud.githubusercontent.com/assets/1515863/23219807/54123ee0-f920-11e6-9d9e-488ddc334151.png)

Commit:
Scroll and max height were set to selectorTableContainer

kdaveed commented:
Scrolling fixed by 230fa73

zarquon42b commented:
Nice fix! However, you should not be able to save an assay without a specimen selectd, which is currently possible.
Reproduce: add assay, do not select bone segment and save, you should be issued an error message telling you to select a bone segment, otherwise adding a measurement datum does not make sense and it is very prone to user error.

zarquon42b commented:
Also, the labels of all instances generated during an investigation are identical, it would be useful to prefix the investigation label, e.g. myinvestigation1.StudyDesignExecution.FrSexEst , so you can easily sort and browse them by their labels.

kdaveed commented:
> However, you should not be able to save an assay without a specimen selectd, which is currently possible.
Reproduce: add assay, do not select bone segment and save, you should be issued an error message telling you to select a bone segment, otherwise adding a measurement datum does not make sense and it is very prone to user error.

The validation of the form is as well topic of the thesis so it will be implemented.

Commit:
Added new field for label of the study design execution.

Commit:
SPARQLDataGetter has the field WebappConnector too. Accordinly the
constructors are modified.

Commit:
VIVOWebappConnector now sends the executed queries to the client
for debug.

Commit:
AJAXController puts the executed queries into the response JSON object.

Commit:
JS library can now display literal field based on config data.

Commit:
StudyDesignExecution entry form generator dirty hack

Commit:
EditConfigurationVTwo handles FormConfig object.

Commit:
If the formelement is not a subformAdder then descriptor key
is not the predicate but the dataKey of the form.

Commit:
The queries are added to the response as well by the FormDataLoader
routine.

Commit:
The textBox routine by the html library can handle input class.

Commit:
JSONLib function

Commit:
LabelDataGetter is adopted to the changes of the WebappConnector.

Commit:
Graph and GraphLib are modified so that they can generate the
labels automatically by concatenating the inputfield's value
with the class label.

kdaveed commented:
@zarquon42b 
783afc1 fixes #92

Commit:
Image label setting works on the form for property isDepicted.

Commit:
Removed custom profile page definition.

Commit:
Missing files

Commit:
Triple configuration corrections. Now the study design execution
is connected directly to the assay.

zarquon42b commented:
@kdaveed : as we discussed yesterday, can you pls split this branch into smaller feature branches, that can be handled more easily? If you issue a pull request, pls also state whether deployment needs a database wipe.

kdaveed commented:
@zarquon42b  : The only commit that addresses n3 config files is 03d4a08. I did not wipe the whole database just for this, but by the __Add/Remove RDF Data__ menu on the admin panel I removed the following n3 file:

[customProfilePageRemove.txt](https://github.com/RDFBones/RDFBones/files/816211/customProfilePageRemove.txt)

(I could attach only .txt but VIVO can read it as well if you select N3)

![screen shot 2017-03-03 at 09 38 37](https://cloud.githubusercontent.com/assets/7580884/23544001/3b9b8ea8-fff5-11e6-98d3-d6ffb4583365.png)

Otherwise the branch contains only the global label feature (the value of one label field is used for all new instances as a prefix) for the study design execution, and some small modifications. So it is not really reasonable to split into smaller branches.

kdaveed commented:
@zarquon42b Now this pull request is not necessary any more, because all of its commits are on the other new pull requests.
